### PR TITLE
Editing and deleting assignment as teacher

### PR DESCRIPTION
**Pull request by Podgore**

### CODE REVIEW

File: StudyHub/StudyHub.BLL/Services/Assignments/AssignmentTaskService.cs [hunk 1]
```diff
@@ -63,6 +63,11 @@ public async Task<List<AssignmentTaskDTO>> GetAssignmentTaskAsync(Guid assignmen
 
     public async Task<AssignmentTaskDTO> UpdateAssignmentTaskAsync(Guid assignmentTaskId, UpdateAssignmentTaskDTO assignmentTask)
     {
+        var assignment = await _assignmentRepository.SingleAsync(assignment => assignment.Tasks.FirstOrDefault(task => task.Id == assignmentTaskId)!.Id == assignmentTaskId);
```
Comment (fedyafedko) [inline StudyHub/StudyHub.BLL/Services/Assignments/AssignmentTaskService.cs:66]: Why u use SingleAsync?

**Review by fedyafedko — CHANGES_REQUESTED** [review]

File: StudyHub/StudyHub.BLL/Services/Assignments/AssignmentTaskService.cs (hunk 1 above)
Comment (aquaprogit) [reply StudyHub/StudyHub.BLL/Services/Assignments/AssignmentTaskService.cs:66]: I think whole search is incorrect, discussed with Ivan, he will check

File: StudyHub/StudyHub.BLL/Services/Assignments/OptionsService.cs [hunk 2]
```diff
@@ -30,18 +30,6 @@ public async Task<List<TaskOptionDTO>> AddTaskOptionsAsync(Guid taskVariantId, L
         return _mapper.Map<List<TaskOptionDTO>>(options);
     }
 
-    public async Task<TaskOptionDTO> UpdateTaskOptionsAsync(Guid optionId, UpdateTaskOptionDTO taskOptions)
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/Assignments/OptionsService.cs:33]: why should we remove it here?

File: StudyHub/StudyHub/Validators/AssignmentValidators/CreateAssignmentValidator.cs [hunk 3]
```diff
@@ -15,12 +15,12 @@ public CreateAssignmentValidator()
 
         RuleFor(x => x.OpeningDate)
             .NotEmpty().WithMessage("OpeningDate is required.")
-            .GreaterThan(DateTime.Now);
+            .GreaterThan(DateTime.Now).WithMessage($"OpeningDate must be greater than {DateTime.Now}");
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Validators/AssignmentValidators/CreateAssignmentValidator.cs:18]: replace {DateTime.Now} with simple "now" mentioning

File: StudyHub/StudyHub/Validators/AssignmentValidators/CreateAssignmentValidator.cs [hunk 4]
```diff
@@ -15,12 +15,12 @@ public CreateAssignmentValidator()
...
 
         RuleFor(x => x.ClosingDate)
             .NotEmpty().WithMessage("ClosingDate is required.")
-            .GreaterThan(DateTime.Now)
-            .GreaterThan(x => x.OpeningDate);
+            .GreaterThan(DateTime.Now).WithMessage($"ClosingDate must be greater than {DateTime.Now}")
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Validators/AssignmentValidators/CreateAssignmentValidator.cs:22]: same here

**Review by aquaprogit — CHANGES_REQUESTED** [review]

File: StudyHub/StudyHub.BLL/Services/UserInvitationService.cs [hunk 5]
```diff
@@ -106,7 +106,14 @@ public async Task<bool> InviteManyAsync(Guid userId, InviteUsersRequest request)
 
             var invitedUser = _mapper.Map<InvitedUser>(registration);
 
+            string dog = "dog";
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/UserInvitationService.cs:109]: ?

Comment (Podgore) [reply StudyHub/StudyHub.BLL/Services/UserInvitationService.cs:109]: Oppssss, sorry, fix it))

File: StudyHub/StudyHub.BLL/Services/UserInvitationService.cs [hunk 6]
```diff
@@ -107,6 +107,11 @@ public async Task<bool> InviteManyAsync(Guid userId, InviteUsersRequest request)
             var invitedUser = _mapper.Map<InvitedUser>(registration);
 
             invitedUsers.Add(invitedUser);
+            invitedUsers.ForEach(i =>
+            {
+                i.Id = userId;
+                i.Email = email;
+            });
```
Comment (fedyafedko) [inline StudyHub/StudyHub.BLL/Services/UserInvitationService.cs:114]: why is that?

**Review by fedyafedko — CHANGES_REQUESTED** [review]

**Review by fedyafedko — APPROVED** [review]

**Review by aquaprogit — APPROVED** [review]

**Pull request merged**